lib/_stream_wrap.js

```diff
@@ -49,7 +50,8 @@ function StreamWrap(stream) {
       this.pause();
       this.removeListener('data', ondata);
 
-      self.emit('error', new Error('Stream has StringDecoder'));
+      self.emit('error', new errors.Error('ERR_STREAM_HAS_STRINGDECODER',
+                'Stream has StringDecoder'));
```

Comment:
Just move the message to the `internal/errors.js` file where the code is declared, then simplify this to just:

```js
self.emit('error', new errors.Error('ERR_STREAM_HAS_STRINGDECODER'));
```